org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/dropwizard/xtext/validation/ValidationMarkerMap.xtend

```diff
@@ -24,7 +24,13 @@ class ValidationMarkerMap {
 	def void updateMarkers(Iterable<ValidationSummary> summaries) {
 		if (!summaries.empty) {
 			val currentUpdate = futureUpdate
-			summaries.forEach[summary|validationMarkers.put(summary.path, summary)]
+			summaries.forEach [ summary |
+				if (summary.hasNoIssues) {
+					validationMarkers.remove(summary.path)
+				} else {
+					validationMarkers.put(summary.path, summary)
+				}
+			]
```

Comment:
This :arrow_up: is not strictly necessary. If removed, the map will remember entries with empty validation summaries (i.e. errors, warnings, infos all zero). That would steadily grow the map and cause more data to be transferred on validation marker requests.